### ISSUE TITLE
Error Downloading When Using Quotes in Filename

### ISSUE THREAD

**Issue by Yowlen** [TeamNewPipe/NewPipe#2229]
- [X] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [X] I checked if the issue/feature exists in the latest version.
- [ ] I did use the [incredible bugreport to markdown converter](https://teamnewpipe.github.io/CrashReportToMarkdown/) to paste bug reports.

When downloading a video with quotation marks in the final filename, NewPipe throws an error. Specifically:
`Code: 1002
java.io.IOException: Invalid argument.`

Removing the quotes from the final filename lets it download perfectly. This is obviously as a result of quotes not getting the proper escape character to let them pass as part of the filename argument, but they shouldn't need to be exempted anyway since both Windows and *nix filesystems don't support quotation marks in the filenames.

The real issues here are:
1. Quotes need to be put on the blacklist for filenames and set to automatically replace them with underscores like slashes, backslashes, and other invalid characters.
2. Some sort of detection method needs to be implemented to prevent users from accidentally entering an invalid character into the filename and thereby causing this error which most people would be unable to make sense of.

**Comment by burneracc69** [comment]
Hello, I just tested this, I was able to download a video with "quotes" like that in them. No problem, did you change the filename or was it with " originally?

I just tested both options, changing filename to ("bla bla" bla bla) and download a video with them from the start. I got no errors whatsoever. Lol

Are you using 0.16.1?

**Comment by burneracc69** [comment]
I tried using different quotation marks. ‘ ’  “ ”  ' '  " ". I downloaded a video with these put there by me. I got zero errors. Nothing. Then what quotation marks did you use? And did you put them there yourself before downloading or were they there from the start?

**Comment by Yowlen** [comment]
1. I'm using the latest on F-Droid, 0.16.1, yes. I wouldn't have checked the box saying I made sure it was in the latest one otherwise.

2. I'm using an Android 7.0, TouchWiz-based ROM, so maybe Samsung's junk or the specific version of Android might be messing things up.

3. The video I was trying to download already had quotes in it. Here's the link, if it helps: https://www.youtube.com/watch?v=2yIELWjG8Tc

4. As stated, the error message in question, as well as the solution I found, indicates the quotes aren't getting escaped properly (`"` vs `\"` for bash script, for example; idk the Java equivalent since idk how to code in Java) and additional, invalid arguments are being added as a result.

5. Other people saying it's not happening for them is ultimately pointless by itself. Give specs so we can compare them or something because it *is* happening for me, and it needs to be resolved. And the only way to do that is to narrow down the cause.
